docs/aws.md

```diff
@@ -6,21 +6,34 @@
 
 Nextflow uses the [AWS security credentials](https://docs.aws.amazon.com/general/latest/gr/aws-sec-cred-types.html) to make programmatic calls to AWS services.
 
-You can provide your AWS access keys using the standard AWS variables shown below:
+The AWS credentials are selected from the following sources, in order of descending priority:
 
-- `AWS_ACCESS_KEY_ID`
-- `AWS_SECRET_ACCESS_KEY`
-- `AWS_DEFAULT_REGION`
+1. Nextflow configuration file - `aws.accessKey` and `aws.secretKey`. See {ref}`AWS configuration<config-aws>` for more details.
 
-If `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` are not defined in the environment, Nextflow will attempt to
-retrieve credentials from your `~/.aws/credentials` and `~/.aws/config` files. The `default` profile can be
-overridden via the environmental variable `AWS_PROFILE` (or `AWS_DEFAULT_PROFILE`).
+2. A custom profile in `~/.aws/credentials` and/or `~/.aws/config`. The profile can be supplied from the `aws.profile` config option, or the `AWS_PROFILE` or `AWS_DEFAULT_PROFILE` environmental variables.
 
-Alternatively AWS credentials and profile can be specified in the Nextflow configuration file. See {ref}`AWS configuration<config-aws>` for more details.
+3. Environment variables - `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, and `AWS_SESSION_TOKEN`.
 
-:::{note}
-Credentials can also be provided by using an IAM Instance Role. The benefit of this approach is that it spares you from managing/distributing AWS keys explicitly. Read the [IAM Roles](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html) documentation and [this blog post](https://aws.amazon.com/blogs/security/granting-permission-to-launch-ec2-instances-with-iam-roles-passrole-permission/) for more details.
-:::
+4. Java System Properties - `aws.accessKeyId`, `aws.secretKey`, and `aws.sessionToken`.
```

Comment:
This is supported indirectly but I don't think it makes sense to mention in the docs